crates/vsock/src/vhu_vsock_thread.rs

```diff
@@ -504,31 +548,14 @@ impl VhostUserVsockThread {
 
             let vring = vring.clone();
             let event_idx = self.event_idx;
-
-            self.pool.spawn_ok(async move {
-                // TODO: Understand why doing the following in the pool works
```

Comment:
Done! I added some comments about current impl and TODO about global thread pool.

Reply:
IIRC the previous TODO was about why we needed another thread to process the packets and not directly in the handler. I don't remember the details, but it seems to me that the student wanted to avoid the thread entirely, partly because the backend should already allocate a thread for the device.

Reply:
Got it, so I just kept the original TODO, and added some comment about thread pool.